utils/UUID.hh

```diff
@@ -206,7 +206,47 @@ inline std::strong_ordering uuid_tri_compare_timeuuid(bytes_view o1, bytes_view
     return res;
 }
 
-}
+template <typename T>
+class UUID_class {
+    int64_t most_sig_bits;
+    int64_t least_sig_bits;
```

Comment:
Why not store UUID instance?

Reply:
Because it is not trivially-default-constructible (it initializes the members to 0)
and so it can't use the serialization templates for trivial types.
To reduce the churn, I just followed the existing `counter_id` approach which is to mimic a UUID and provide conversion methods.

There's a static_assert that checks that these are POD types.